### PR TITLE
Expand Vanilla Xp drops color range

### PR DESCRIPTION
**Pull request by TzZek**
Xp drop plugin now further lets you customize the vanilla xp drops by
having access to the full RGB colors instead of the default ones found
in the xp drop orb. Also added positions to the config file for experience 
drops.

idea64_2018-05-21_14-46-24

### CODE REVIEW

**Comment by YezziRS** [comment]
> Also added positions to the config file for experience
drops.

Does this mean this comes with `CENTER_CANVAS` positions or?

**Comment by YezziRS** [comment]
@TzZek Yeah watched the code, new to the github thing and open source, sorry for the question, awesome job btw! :-)

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 1]
```diff
@@ -30,49 +30,79 @@
 import net.runelite.client.config.ConfigItem;
 
 @ConfigGroup(
-	keyName = "xpdrop",
-	name = "XP Drop",
-	description = "Configuration for XP drop customization"
+		keyName = "xpdrop",
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Indent is too great here

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 2]
```diff
@@ -30,49 +30,79 @@
...
+		name = "XP Drop",
+		description = "Configuration for XP drop customization"
 )
 public interface XpDropConfig extends Config
 {
 	@ConfigItem(
-		keyName = "hideSkillIcons",
-		name = "Hide skill icons",
-		description = "Configure if XP drops will show their respective skill icons"
+			keyName = "hideSkillIcons",
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 3]
```diff
@@ -30,49 +30,79 @@
...
 	default boolean hideSkillIcons()
 	{
 		return false;
 	}
 
 	@ConfigItem(
-		keyName = "meleePrayerColor",
-		name = "Melee Prayer Color",
-		description = "XP drop color when a melee prayer is active"
+			keyName = "meleePrayerColor",
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 4]
```diff
@@ -30,49 +30,79 @@
...
 	default Color getMeleePrayerColor()
 	{
 		return new Color(0x15, 0x80, 0xAD);
 	}
 
 	@ConfigItem(
-		keyName = "rangePrayerColor",
-		name = "Range Prayer Color",
-		description = "XP drop color when a range prayer is active"
+			keyName = "rangePrayerColor",
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 5]
```diff
@@ -30,49 +30,79 @@
...
 public interface XpDropConfig extends Config
 {
 	@ConfigItem(
-		keyName = "hideSkillIcons",
-		name = "Hide skill icons",
-		description = "Configure if XP drops will show their respective skill icons"
+			keyName = "hideSkillIcons",
+			name = "Hide skill icons",
+			description = "Configure if XP drops will show their respective skill icons",
+			position = 6
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: The config options should be in order in this file

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 6]
```diff
@@ -30,49 +30,79 @@
...
 	default Color getRangePrayerColor()
 	{
 		return new Color(0x15, 0x80, 0xAD);
 	}
 
 	@ConfigItem(
-		keyName = "magePrayerColor",
-		name = "Mage Prayer Color",
-		description = "XP drop color when a mage prayer is active"
+			keyName = "magePrayerColor",
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 7]
```diff
@@ -30,49 +30,79 @@
...
+			description = "XP drop color when a mage prayer is active",
+			position = 3
 	)
 	default Color getMagePrayerColor()
 	{
 		return new Color(0x15, 0x80, 0xAD);
 	}
+
+	@ConfigItem(
+			keyName = "defaultXpDrop",
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 8]
```diff
@@ -30,49 +30,79 @@
...
+			name = "Regular XP Drop",
+			description = "More custom default colors",
+			position = 5
+	)
+	default Color getDefaultXpDrop()
+	{
+		return new Color(0x15, 0x80, 0xAD);
+	}
+	@ConfigItem(
+			keyName = "overrideVanillaColor",
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java [hunk 9]
```diff
@@ -30,49 +30,79 @@
...
+			position = 4
+
+	)
+	default boolean overrideVanillaColor()
+	{
+		return false;
+	}
 }
+
+
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropConfig.java]: Remove extra new lines

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropPlugin.java [hunk 10]
```diff
@@ -41,7 +41,7 @@
 import net.runelite.client.plugins.PluginDescriptor;
 
 @PluginDescriptor(
-	name = "XP Drop"
+		name = "XP Drop"
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropPlugin.java]: Indent

File: runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropPlugin.java [hunk 11]
```diff
@@ -122,13 +122,22 @@ public void onWidgetHidden(WidgetHiddenChanged event)
 
 			widget.setTextColor(color);
 		}
+
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/experiencedrop/XpDropPlugin.java]: Remove extra new line

**Comment by TzZek** [comment]
Any chance of this getting reviewed again ^_^

**Comment by Adam-** [comment]
Sorry we apparently ignored this for like 2 years. If you want to rebase this it would be simplier now since we have a way to specify optional Color configs. Ping me if you do so.

**Comment by TzZek** [comment]
> Sorry we apparently ignored this for like 2 years. If you want to rebase this it would be simplier now since we have a way to specify optional Color configs. Ping me if you do so.

No biggy haha, just was cleaning up my repos / pulls / etc. RuneLite has always been a very busy project but ya'll always stayed dedicated. Figured it'd be time to retire this. <3

**Pull request closed without merging**